src/main/java/chessmaster/game/Game.java

```diff
@@ -66,11 +67,11 @@ public Game(Color playerColour, Color currentTurnColor, ChessBoard board,
         assert (1 <= difficulty) && (difficulty <= 3) : "Difficulty should be between 1 and 3!";
     }
 
-    public void run() {
+    public boolean run() {
         ui.printText(START_HELP_STRINGS);
         ui.printChessBoard(board.getBoard());
 
-        while (!hasEnded && !AbortCommand.isAbortCommand(command)) {
+        while (!hasEnded && !AbortCommand.isAbortCommand(command) && !RestartCommand.isRestartCommand(command)) {
```

Comment:
Help me rename the function to `isAbort` instead of `isAbortCommand`. Same for `isRestart` to keep things short and sweet